tests/python/contrib/test_hexagon/test_conv2d_conv2d.py

```diff
@@ -0,0 +1,340 @@
+    )
+
+    # Calculate packed input
+    packed_shape = get_packed_activation_layout(X_pad.shape, block_shape)
+    X_packed = te.compute(
+        packed_shape,
+        lambda n, ho, wo, co, hi, wi, ci: X_pad[
+            n, ho * block_H + hi, wo * block_W + wi, co * block_C + ci
+        ],
+        name="packed_input",
+    )
+
+    filter_Cio, filter_Ki, filter_Cii = get_filter_block_shape()
+    filter_Ci = filter_Cio * filter_Cii
+
+    rh = te.reduce_axis((0, kernel_size1[0]), name="rh")
+    rw = te.reduce_axis((0, kernel_size1[1]), name="rw")
+    rc = te.reduce_axis((0, C), name="rc")
+
+    def compute(n, ho, wo, ko, hi, wi, ki):
+        h = ho * block_H + hi
+        h_contig = h * stride1[0] + rh
+        h_block_id = h_contig // block_H
+        h_block_offset = h_contig % block_H
+
+        w = wo * block_W + wi
+        w_contig = w * stride1[1] + rw
+        w_block_id = w_contig // block_W
+        w_block_offset = w_contig % block_W
+
+        c_block_id = rc // block_C
+        c_block_offset = rc % block_C
+
+        rco = rc // filter_Ci
+        rcio = (rc % filter_Ci) // filter_Cii
+        rcii = rc % filter_Cii
+
+        return te.sum(
+            X_packed[
+                n,
+                h_block_id,
+                w_block_id,
+                c_block_id,
+                h_block_offset,
+                w_block_offset,
+                c_block_offset,
+            ]
+            * filt_packed1[ko, rco, rh, rw, rcio, ki, rcii],
+            axis=[rh, rw, rc],
+        )
+
+    output_shape1 = get_packed_activation_layout(logical_output_shape1, block_shape)
+    temp_Y = te.compute(output_shape1, compute, name="temp_output")
+
+    # calculate kernel size and output channels
+    # given oihw8i32o4i filter layout
+    kernel_size2 = tuple(shape_filter2[2:4])
+    out_channels2 = shape_filter2[0] * shape_filter2[5]
+
+    # get the the logical output shape of conv2d #2
+    logical_input_shape2 = logical_output_shape1
+    logical_output_shape2 = get_conv2d_nhwc_shape(
+        logical_input_shape2,
+        kernel_size2,
+        stride2,
+        pad2,
+        dilation2,
+        out_channels2,
+    )
+
+    rh = te.reduce_axis((0, kernel_size2[0]), name="rh")
+    rw = te.reduce_axis((0, kernel_size2[1]), name="rw")
+    rc = te.reduce_axis((0, logical_input_shape2[3]), name="rc")
+
+    def compute2(n, ho, wo, ko, hi, wi, ki):
+        h = ho * block_H + hi
+        h_contig = h * stride2[0] + rh
+        h_block_id = h_contig // block_H
+        h_block_offset = h_contig % block_H
+
+        w = wo * block_W + wi
+        w_contig = w * stride2[1] + rw
+        w_block_id = w_contig // block_W
+        w_block_offset = w_contig % block_W
+
+        c_block_id = rc // block_C
+        c_block_offset = rc % block_C
+
+        rco = rc // filter_Ci
+        rcio = (rc % filter_Ci) // filter_Cii
+        rcii = rc % filter_Cii
```

Comment:
nit: Consider factoring out the index calculation into a separate helper function that can be used in both compute1 and compute2.

Reply:
Yes.  I would like to do this in a refactor which is already work in progress.